Source/WebKit/GPUProcess/graphics/WebGPU/RemoteGPU.cpp

```diff
@@ -75,10 +75,15 @@ void RemoteGPU::stopListeningForIPC(Ref<RemoteGPU>&& refFromConnection)
 {
     assertIsMainRunLoop();
     m_streamConnection->invalidate();
-    m_streamConnection->stopReceivingMessages(Messages::RemoteGPU::messageReceiverName(), m_identifier.toUInt64());
-    workQueue().dispatch([protectedThis = WTFMove(refFromConnection)]() {
-        protectedThis->workQueueUninitialize();
+
+    // This item is dispatched to the WorkQueue such that it will process it last, after any existing work.
```

Comment:
You're adding the invalidate twice.
Your commit message describes what the code does, in literal sense. This is also what the code changes describe.
However, hunks like these would benefit from describing what is the purpose, why the code does what it does.